File: python/convert_an_array_into_2d_array_with_conditions.py
# no. 2610
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        ans = []
        curr_row = {}
        for n in nums:
            row = curr_row.get(n, 0)
            if row > len(ans) - 1:
                ans.append([n])
            else:
                ans[row].append(n)
            curr_row[n] = row + 1
        return ans

File: C++/convert_an_array_into_2d_array_with_conditions.cpp
// no. 2610
#include <map>
#include <vector>
using namespace std;
class Solution {
public:
  vector<vector<int>> findMatrix(vector<int> &nums) {
    vector<vector<int>> ans = {};
    map<int, int> curr_row;
    for (int i = 0; i < nums.size(); i++) {
      int row = curr_row[nums[i]];
      int n_rows = ans.size();
      if (row > n_rows - 1) {
        vector<int> new_row;
        new_row.push_back(nums[i]);
        ans.push_back(new_row);
      } else {
        ans[row].push_back(nums[i]);
      }
      curr_row[nums[i]] = row + 1;
    }
    return ans;
  }
};
